include/tvm/runtime/data_type.h

```diff
@@ -110,7 +111,7 @@ class DataType {
     return -lanes_as_int;
   }
   /*! \return whether type is a scalar type. */
-  bool is_scalar() const { return lanes() == 1; }
+  bool is_scalar() const { return !is_scalable_vector() && lanes() == 1; }
```

Comment:
I think it would make more sense to use `!is_scalable_or_fixed_length_vector()` here. The current version is going to throw an error if applied on a scalable vector.

Reply:
Ok yes, thinking about it, this probably is the most solid way to do it, even though it reads a bit awkward 👍